### PR TITLE
pkg/api/handlers/libpod: close quadlets multipart files per iteration and prevent duplicates

### LINKED ISSUE #29752: pkg/api/handlers/libpod: quadlets multipart differs from manifests
`processMultipartQuadlets` and the manifests upload handler both read a client multipart stream and write each part into a directory, but they handle it differently.

`manifests.go` wraps each part in a closure so `part.Close()` and the file close run per iteration, with a comment saying that is deliberate. `quadlets.go` defers both inside the loop, so a request with many parts holds every handle open until the function returns.

`manifests.go` also opens with `O_CREATE|O_EXCL|O_WRONLY`, so a repeated filename is an error. `quadlets.go` uses `os.Create`, so a second part named the same as an earlier one silently overwrites it.

Should quadlets follow manifests here, and should a duplicate filename be an error or documented as last-wins?

### PR DESCRIPTION
**Pull request by madhoshyagnik**
#### Description
`processMultipartQuadlets` previously deferred closing each multipart part and file inside the loop, leaking descriptors until the handler returned. It also opened files with `os.Create`, silently overwriting earlier files if a request contained parts with duplicate names.

This PR aligns `processMultipartQuadlets` with the manifests multipart upload handler:
- Wrap each part processing in a closure so part and file descriptors close immediately on each loop iteration.
- Use `os.OpenFile` with `os.O_CREATE|os.O_EXCL|os.O_WRONLY` (`0600`) so duplicate filenames return an error instead of silently overwriting.
- Sanitize filenames using `filepath.Base` to prevent directory traversal.
- Add unit tests verifying multiple file extraction, duplicate name rejection, and path traversal sanitization.

Fixes: #29752

#### Checklist

Ensure you have completed the following checklist for your pull request to be reviewed:

- [x] I have read and understood our [contributing guidelines](https://github.com/podman-container-tools/podman/blob/main/CONTRIBUTING.md) and will not have more than two open PRs as a new contributor.
- [x] PR description, commit message, and GitHub comments are human-written, per [LLM Policy](https://github.com/podman-container-tools/community/blob/main/LLM_POLICY.md)
- [x] Certify you wrote the patch or otherwise have the right to pass it on as an open-source patch by signing all commits. (`git commit -s`). The author email must match the sign-off email address.
- [x] Referenced issues using `Fixes: #00000` in commit message (if applicable)
- [x] [Tests](https://github.com/podman-container-tools/podman/tree/main/test#readme) have been added/updated (or no tests are needed)
- [x] [Documentation](https://github.com/podman-container-tools/podman/blob/main/docs/README.md) has been updated (or no documentation changes are needed)
- [x] All commits pass `make validatepr` (format/lint checks)
- [x] [Release note](https://github.com/kubernetes/community/blob/master/contributors/guide/release-notes.md) entered in the section below (or `None` if no user-facing changes)

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by Honny1 — CHANGES_REQUESTED** [review]
Code LGTM, please fix this issue:

```
hack/commit-subject-check.sh be177b24343f278db9cf0c01fac05cc680951e7f..HEAD
c1fe77dcc04284cb42fd9a521a1554c26afea535: subject is 92 chars (must be < 90): pkg/api/handlers/libpod: close quadlets multipart files per iteration and prevent duplicates
```

**Comment by madhoshyagnik** [comment]
Hi @Honny1 and @TomSweeneyRedHat, thanks for the review! I’ve made the requested changes. Please take another look when you get a chance. Thanks!

File: pkg/api/handlers/libpod/quadlets.go [hunk 1]
```diff
@@ -136,25 +136,37 @@ func processMultipartQuadlets(tempDir string, r *http.Request) ([]string, error)
...
 			continue
 		}
-
-		// Create file in temp directory
-		filePath := filepath.Join(quadletDir, filename)
-		file, err := os.Create(filePath)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create file %s: %w", filename, err)
+		filename = filepath.Base(filename)
+		if filename == "." || filename == string(filepath.Separator) {
```
Comment (Honny1) [inline pkg/api/handlers/libpod/quadlets.go]: What if`filename == ".."`?

Comment (madhoshyagnik) [reply pkg/api/handlers/libpod/quadlets.go]: Thanks for catching that! Updated to skip `..` as well.

**Review by Honny1 — APPROVED** [review]
LGTM

**Comment by mheon** [comment]
Sure, LGTM
